src/main/java/ogc/rs/apiserver/ApiServerVerticle.java

```diff
@@ -1433,16 +1430,27 @@ public void getStacItemById(RoutingContext routingContext){
     dbService
         .getStacItemById(stacCollectionId, stacItemId)
         .onSuccess(stacItem -> {
-              JsonObject assets = new JsonObject();
               try {
-                    JsonArray allLinksInFeature = new JsonArray(commonLinksInFeature.toString());
+                // Retrieve user authentication info
+                AuthInfo userKey = routingContext.get(USER_KEY);
+                long expiry = (userKey != null) ? userKey.getExpiry() : 0;
+
+                // Check if token is expired
+                boolean isExpired = Instant.now().getEpochSecond() >= expiry;
+                boolean shouldCreate = routingContext.get("shouldCreate");
+                shouldCreate = !isExpired && shouldCreate;
+
+                if (isExpired) {
+                  LOGGER.warn("Token expired. Disabling pre-signed URL generation.");
+                }
```

Comment:
- Instead of having this check here, let's move it to the handler. We can check if the expiry of the token is less than some amount of seconds away (like maybe 10 seconds). If it is, `shouldCreate` can be set to false.
- We can also add a detail about this in the API documentation, that if the token expiry is less than 10 seconds away, presigned URLs will not be generated.